trees.h

```diff
@@ -3,7 +3,7 @@
 
 /* header created automatically with maketrees.c */
 
-ZLIB_INTERNAL const ct_data static_ltree[L_CODES+2] = {
+const ct_data ZLIB_INTERNAL static_ltree[L_CODES+2] = {
```

Comment:
Why these changes?
I know this is technically the correct order, but GCC doesn't care, we don't do this other places and does this really belong in this PR?
My personal opinion on this is to put these flags at the start of the line for easier readability, and for some attributes even on the preceding line to avoid cluttering it up with stuff that is ultimately not interesting (like HOT, COLD, ALLOC, etc).

Reply:
I'm not so concerned about the order. I had to change `static_dtree` and `base_length` from static to non-static. And I noticed this discrepancy. This instance was the only place where it had this order. There were two other instances in the same file with the const first order.

Reply:
Let me know how you want it and I can change it.

Reply:
Lets go with attributes-first, it does make things more readable.